src/java/main/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -117,13 +117,21 @@ private QuorumServer(long id, InetSocketAddress addr,
             this.id = id;
             this.addr = addr;
             this.electionAddr = electionAddr;
+            String ipReachableValue = System.getProperty("zookeeper.ipReachableTimeout");
```

Comment:
This code is duplicated four times, not good :) - you can pull this into a dedicated function instead. Also make sure to add try catch around the parseInt to prevent crash against illegal input values.